src/openvic-simulation/dataloader/Dataloader.cpp

```diff
@@ -0,0 +1,288 @@
+#include "Dataloader.hpp"
+
+#include "openvic-simulation/GameManager.hpp"
+#include "openvic-simulation/utility/Logger.hpp"
+
+#include <openvic-dataloader/csv/Parser.hpp>
+#include <openvic-dataloader/detail/CallbackOStream.hpp>
+#include <openvic-dataloader/v2script/Parser.hpp>
+
+using namespace OpenVic;
+using namespace OpenVic::NodeTools;
+using namespace ovdl;
+
+bool Dataloader::set_roots(std::vector<std::filesystem::path> new_roots) {
+	if (!roots.empty()) {
+		Logger::error("Overriding existing dataloader roots!");
+		roots.clear();
+	}
+	bool ret = true;
+	for (std::reverse_iterator<std::vector<std::filesystem::path>::const_iterator> it = new_roots.crbegin(); it != new_roots.crend(); ++it) {
```

Comment:
I do wonder if it might not be more efficient to give long STL types a shortened type alias in the namespace, so say `std::vector<std::filesystem::path>` becomes `path_vector` so this could be replaced with `std::reverse_iterator<path_vector::const_iterator>` if we'll avoid auto for this. (I'd probably foremost suggest path itself be made shortened to something like `fs_path` or `fs::path` anyway)

Reply:
Good ideas 👍